### PR TITLE
Extend system_settings role

### PR DESCRIPTION
**Pull request by machacekondra**
Extend system_settings role by:
- Proxy
- DNS

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,4 +8,4 @@ install-ansible-collections:
 
 .PHONY: integration
 integration: install-python-packages install-ansible-collections
-	ansible-test integration --no-temp-workdir
+	ansible-test integration --no-temp-workdir system_settings_test
```
Comment (bardielle) [inline Makefile:11]: please revert this change

Comment (machacekondra) [reply Makefile]: Done

File: roles/system_settings/tasks/dns.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+- name: Append the DNS servers
```
Comment (bardielle) [inline roles/system_settings/tasks/dns.yml:1]: maybe we need to split this task into 3 blocks:  DNS servers, DNS domains, and DNS hostname, what do you think?

File: roles/system_settings/tasks/dns.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  vmware.vmware_rest.appliance_networking_dns_servers:
```
Comment (bardielle) [inline roles/system_settings/tasks/dns.yml:2]: maybe we need to split this task into 3 blocks: DNS servers, DNS domains, and DNS hostname, what do you think?

Comment (machacekondra) [reply roles/system_settings/tasks/dns.yml:2]: What do you mean by blocks?

Comment (bardielle) [reply roles/system_settings/tasks/dns.yml:2]: I think it will be more clear like that: 
```
- name: DNS servers
  when: system_settings_dns_mode is defined or system_settings_dns_servers is defined
  block:
  - name: Append the DNS servers
    ...
    when: system_settings_dns_mode_append
  - name: Set the DNS servers
    ...
    when: system_settings_dns_mode_append 
    ...

- name: DNS domains
  when: system_settings_dns_domains
  block:
  - name: Append DNS domains
    ...
    when: system_settings_dns_mode_append
  - name: Set the DNS domains
    ...
    when: not system_settings_dns_mode_append
    ...
```

Comment (machacekondra) [reply roles/system_settings/tasks/dns.yml]: Yeah, that looks definitely, nicer, thanks!

**Review by bardielle — APPROVED** [review]
/lgtm
/approve
